src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/Designer/LaunchProfileEditorServiceFactory.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.ServiceHub.Framework.Services;
+using Microsoft.VisualStudio.ProjectSystem.Debug.Designer;
+using Microsoft.VisualStudio.RpcContracts;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug.Designer
+{
+    [Export(typeof(IPackageService))]
+    internal sealed class LaunchProfileEditorServiceFactory : IPackageService, IDisposable
+    {
+        private const string LaunchProfileEditorServiceName = "Microsoft.VisualStudio.ProjectSystem.Managed.LaunchProfileEditorService";
+        private const string ProjectGuidActivationArgumentName = "ProjectGuid";
+
+        internal static readonly ServiceJsonRpcDescriptor LaunchProfileEditorServiceDescriptorV1 = new(
+            new ServiceMoniker(LaunchProfileEditorServiceName, new Version(0, 1)),
+            ServiceJsonRpcDescriptor.Formatters.UTF8,
+            ServiceJsonRpcDescriptor.MessageDelimiters.HttpLikeHeaders);
+
+        private readonly IProjectServiceAccessor _projectServiceAccessor;
+        private IDisposable? _profferDisposable;
+        private ILaunchProfileEditorService? _launchProfileEditorService;
+
+        [ImportingConstructor]
+        public LaunchProfileEditorServiceFactory(IProjectServiceAccessor projectServiceAccessor)
+        {
+            _projectServiceAccessor = projectServiceAccessor;
+        }
+
+        public async Task InitializeAsync(IAsyncServiceProvider asyncServiceProvider)
+        {
+            IBrokeredServiceContainer brokeredServiceContainer = await asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+
+            _profferDisposable = brokeredServiceContainer.Proffer(LaunchProfileEditorServiceDescriptorV1, CreateInstanceAsync);
+
+            async ValueTask<object?> CreateInstanceAsync(
+                ServiceMoniker moniker,
+                ServiceActivationOptions options,
+                IServiceBroker serviceBroker,
+                AuthorizationServiceClient authorizationServiceClient,
+                CancellationToken cancellationToken)
+            {
+                await authorizationServiceClient.AuthorizeOrThrowAsync(WellKnownProtectedOperations.CreateClientIsOwner(), cancellationToken);
+
+                if (options.ActivationArguments is null
+                    || !options.ActivationArguments.TryGetValue("ProjectGuid", out string projectGuidString))
```

Comment:
When the client requests the service we expect it to pass the target project's GUID. This is how we figure out which list of launch profiles to provide.